### ISSUE TITLE
behaviour when changing the owner of a group

### ISSUE THREAD

**Issue by joachim-n** [Gizra/og#306]
I create a group. I was automatically added as a member.

I then edited the node, and made another user the node owner. I was expecting a membership to be made for them, but it wasn't.

Another valid behaviour here would be to prevent changing the owner of a group to a user who is not already a member.

**Comment by MPParsley** [comment]
Instead of making someone member I'd go for a constraint. cc @pfrenssen @amitaibu wdyt?

**Comment by amitaibu** [comment]
I thin both cases might have a use case, but I guess having the user first become a member might be more according to users' expectations.

**Comment by pfrenssen** [comment]
Yes I think we should let this be implemented in a way that the project requires. The initial author of the group becomes the administrator, but there can be many administrators and they are not necessarily tied to the group ownership forever. Changing entity ownership is not a common operation, and I even think this is not supported in core. It will always require some custom code or set up with a contrib module like Rules or Views Bulk Operations.

I think the current implementation is possibly fine, we cannot change this in a way that would satisfy all our users. Let's leave this to be implemented by projects how it best fits their own use case. For example in our project there is only one administrator allowed, and an administrator can pass their role to another member when they wish to retire from the group. The group authorship remains with the original author though.

**Comment by pfrenssen** [comment]
I think we maybe should drop the functionality that gives a group owner the full rights to maintain the group forever? There is no real need for this since this the group author is already been given the admin role. I think our idea of placing full trust in the original creator of a group can backfire. We have seen cases where group admins needed to be removed from their groups after complaints about toxic behavior and abuse of their privileges. We can discuss this in a separate issue.
